doc/configuration.md

```diff
@@ -83,9 +83,12 @@ limits.processes            | integer   | - (max)       | yes           | Maximu
 linux.kernel\_modules       | string    | -             | yes           | Comma separated list of kernel modules to load before starting the container
 raw.apparmor                | blob      | -             | yes           | Apparmor profile entries to be appended to the generated profile
 raw.lxc                     | blob      | -             | no            | Raw LXC configuration to be appended to the generated one
+raw.seccomp                 | blob      | -             | no            | Raw LXC seccomp profile string to append to the generated one
 security.nesting            | boolean   | false         | yes           | Support running lxd (nested) inside the container
 security.privileged         | boolean   | false         | no            | Runs the container in privileged mode
 user.\*                     | string    | -             | n/a           | Free form user key/value storage (can be used in search)
+security.syscalls.default   | boolean   | true          | no            | Enables the default syscall blacklist
+security.syscalls.compat    | boolean   | true          | no            | On x86\_64 this enables blocking of compat\_\* syscalls, it is a no-op on other arches
```

Comment:
On Mon, May 23, 2016 at 03:39:28PM -0700, Serge Hallyn wrote:
> >  security.nesting            | boolean   | false         | yes           | Support running lxd (nested) inside the container
> >  security.privileged         | boolean   | false         | no            | Runs the container in privileged mode
> >  user.\*                     | string    | -             | n/a           | Free form user key/value storage (can be used in search)
> > +security.syscalls.default   | boolean   | true          | no            | Enables the default syscall blacklist
> > +security.syscalls.compat    | boolean   | true          | no            | On x86\_64 this enables blocking of compat\_\* syscalls, it is a no-op on other arches
> 
> Hey - thanks for doing this.  But a few comments:
> 
> 1. it's not clear to me whether 'security.syscalls.compat = true' means allow them or block them.

Yes, I figured nobody would like the names I chose so I didn't choose
very good ones :). Thoughts are welcome.

> 2. is just appending a raw.seccomp always enough?  Or does raw.seccomp really need to be a replacement?  I would have actually expected a security.syscalls.whitelist and blacklist which would get turned into a seccomp config for you, and raw.seccomp finally to be a full replacement, ie. security.syscalls.* is ignored if raw.seccomp is set.

raw.apparmor appends, so I was just following that behavior. I could
implement the security.syscalls.{blacklist,whitelist} stuff pretty
simply, though.

can you not have multiple [blacklist] and [whitelist] directives in v2
policies? If so, can't the person configuring raw.seccomp specify
things exactly as they want? (In practice, they can do this anyway
since they could just disable compat and default and specify
raw.seccomp anyway.)

> 
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/lxc/lxd/pull/2041/files/1c9e7f5be2cf4b4aaf518cd085802c364b10b209#r64301786